Update remote_calendar docs for basic auth support (#43965)
home-assistant/home-assistant.github.io: source/_integrations/remote_calendar.markdown

Reviewer: Can you update with the new optional configuration steps you added for username and password?

## Known limitations

> The integration does not provide the ability to connect to a resource that requires special headers or authentication methods other than HTTP Basic Authentication.

Reviewer: The integration supports HTTP Basic Authentication, but does not provide the ability to connect to a resource that requires any other special headers.

> The integration does not provide the ability to connect to a resource that requires special headers or authentication methods other than HTTP Basic Authentication.

Reviewer 2: ```suggestion
The integration supports HTTP Basic Authentication but does not provide the ability to connect to a resource that requires other special headers.
```

  description: "Enable SSL certificate verification when connecting via HTTPS."
{% endconfiguration_basic %}

### Authentication

> If the calendar URL requires HTTP Basic Authentication (the server responds with `401 Unauthorized` and a `WWW-Authenticate: Basic` header), you will be prompted for a username and password in a second step. The credentials are stored in the configuration entry and used for all subsequent calendar fetches.

Reviewer: Agree with co-pilots suggestion here.
